### ISSUE TITLE
Support text files compressed with gzip, bzip2 and xz

### ISSUE THREAD

**Issue by robinsmidsrod** [google/codesearch#59]
It would be awesome to be able to search my IRC log files, but I compress them to save space. Would it be possible to implement automatic decompression of gzip, bzip2 and xz to the toolchain?

**Comment by eborisch** [comment]
Another option would be to store your IRC logs on a compressed file system like ZFS...

**Comment by robinsmidsrod** [comment]
@eborisch Yeah, sure, but ZFS is not easily available on all operating systems, and sometimes you're not in control of which filesystem is used where you store your files. Doing something similar to what less and cat does with zless and zcat would be really nice.
